modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/registries/RegistryLookup.scala

```diff
@@ -80,7 +80,61 @@ object RegistryLookup {
       RegistryLookup[F].list(repositoryRef, vendor: String, name: String, model: Int)
   }
 
-  def inMemoryLookup(
+  /**
+   * An implementation of [[RegistryLookup]] with standard implementations of embedded/in-memory
+   *  lookups and pluggable implementation of http lookups
+   */
+  private[registries] abstract class StdRegistryLookup[F[_]: Sync] extends RegistryLookup[F] {
+
+    /** Abstract methods to be provided by the implementation */
+    def httpLookup(registry: Registry.Http, schemaKey: SchemaKey): F[Either[RegistryError, Json]]
+    def httpList(
+      registry: Registry.Http,
+      vendor: String,
+      name: String,
+      model: Int
+    ): F[Either[RegistryError, SchemaList]]
+
+    /** Common functionality across all implementations */
+    override def lookup(registry: Registry, schemaKey: SchemaKey): F[Either[RegistryError, Json]] =
+      registry match {
+        case http: Registry.Http =>
+          withErrorHandling {
+            httpLookup(http, schemaKey)
+          }
+        case Registry.Embedded(_, path) => Embedded.lookup[F](path, schemaKey)
+        case Registry.InMemory(_, schemas) =>
+          Sync[F].delay(RegistryLookup.inMemoryLookup(schemas, schemaKey))
+      }
+
+    override def list(
+      registry: Registry,
+      vendor: String,
+      name: String,
+      model: Int
+    ): F[Either[RegistryError, SchemaList]] =
+      registry match {
+        case http: Registry.Http =>
+          withErrorHandling {
+            httpList(http, vendor, name, model)
+          }
+        case Registry.Embedded(_, base) =>
+          val path = toSubpath(base, vendor, name)
+          Sync[F].delay(Embedded.unsafeList(path, model))
+        case _ => Sync[F].pure(RegistryError.NotFound.asLeft)
+      }
+
+  }
```

Comment:
Hint to reviewers: start here!

This is the main new code.  Everything else is just shuffling code around, with nothing new.